### PR TITLE
syncval: Clean up execution scope data

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
Add struct SyncExecScope to store the different
forms of execution scope flags so that it is easier
to set and pass them consistently.

Change-Id: I60e15955dafab01fba0183c5d8023f7c016fc761

### CODE REVIEW

File: layers/synchronization_validation.h [hunk 1]
```diff
@@ -96,27 +96,41 @@ struct HazardResult {
              const SyncStageAccessFlags &prior_, const ResourceUsageTag &tag_);
 };
 
+struct SyncExecScope {
```
Comment (jeremyg-lunarg) [inline layers/synchronization_validation.h:99]: If you have a better name for this struct or any of its members, I'm  all ears...

File: layers/synchronization_validation.h [hunk 2]
```diff
@@ -96,27 +96,41 @@ struct HazardResult {
...
+    VkPipelineStageFlags mask_param;
+    VkPipelineStageFlags expanded_mask;
+    VkPipelineStageFlags exec_scope;
+    SyncStageAccessFlags valid_accesses;
+
+    SyncExecScope() : mask_param(0), expanded_mask(0), exec_scope(0), valid_accesses(0) {}
+};
+
+SyncExecScope MakeSrcScope(VkQueueFlags queue_flags, VkPipelineStageFlags src_stage_mask);
```
Comment (jeremyg-lunarg) [inline layers/synchronization_validation.h:108]: same goes for how I did these pseudo-constructors

**Review by jzulauf-lunarg — COMMENTED** [review]
LG -- a couple of requested cleanups.

File: layers/synchronization_validation.cpp [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /* Copyright (c) 2019-2020 The Khronos Group Inc.
  * Copyright (c) 2019-2020 Valve Corporation
- * Copyright (c) 2019-2020 LunarG, Inc.
+ * Copyright (c) 2019-2021 LunarG, Inc.
```
Comment (jzulauf-lunarg) [inline layers/synchronization_validation.cpp:3]: update date on all fields per @mark-lunarg .

File: layers/synchronization_validation.cpp [hunk 4]
```diff
@@ -2304,24 +2295,18 @@ void CommandBufferAccessContext::RecordSetEvent(VkCommandBuffer commandBuffer, V
     // Given:
     //     Stuff1; SetEvent; Stuff2; SetEvent; WaitEvents;
     // WaitEvents cannot know which of Stuff1, Stuff2, or both has completed execution.
+    auto scope = SyncExecScope::MakeSrc(GetQueueFlags(), stageMask);
```
Comment (jzulauf-lunarg) [inline layers/synchronization_validation.cpp:2298]: Nice

File: layers/synchronization_validation.cpp [hunk 5]
```diff
@@ -2919,13 +2900,47 @@ void RenderPassAccessContext::RecordEndRenderPass(AccessContext *external_contex
...
+}
+
+SyncBarrier::SyncBarrier(const SyncExecScope &src, const SyncExecScope &dst) {
+    src_exec_scope = src.exec_scope;
+    src_access_scope = 0;
+    dst_exec_scope = dst.exec_scope;
+    dst_access_scope = 0;
+}
+
+template <typename Barrier>
```
Comment (jzulauf-lunarg) [inline layers/synchronization_validation.cpp:2928]: Much DRY'r

File: layers/synchronization_validation.h [hunk 6]
```diff
@@ -1,6 +1,6 @@
 /*
  * Copyright (c) 2019-2020 Valve Corporation
- * Copyright (c) 2019-2020 LunarG, Inc.
+ * Copyright (c) 2019-2021 LunarG, Inc.
```
Comment (jzulauf-lunarg) [inline layers/synchronization_validation.h:3]: change all date lines

File: layers/synchronization_validation.h (hunk 1 above)
Comment (jzulauf-lunarg) [reply layers/synchronization_validation.h:119]: +100

**Review by jzulauf-lunarg — APPROVED** [review]
Fix the copyright date changes.  then GTG.

**Review by jzulauf-lunarg — APPROVED** [review]
LGTM

**Pull request merged**